packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -1263,7 +1263,8 @@ class _SliverAppBarDelegate extends SliverPersistentHeaderDelegate {
 
     final bool isScrolledUnder = overlapsContent || forceElevated || (pinned && shrinkOffset > maxExtent - minExtent);
     final bool isPinnedWithOpacityFade = pinned && floating && bottom != null && extraToolbarHeight == 0.0;
-    final double toolbarOpacity = !pinned || isPinnedWithOpacityFade
+    final bool isAccessibilityOn = MediaQuery.of(context).accessibleNavigation;
+    final double toolbarOpacity = !isAccessibilityOn &&(!pinned || isPinnedWithOpacityFade)
```

Comment:
Tiny tweak; improve readability just a little:
```suggestion
    final bool accessibilityIsDisabled = !MediaQuery.of(context).accessibleNavigation;
    final double toolbarOpacity =  accessibilityIsDisabled && (!pinned || isPinnedWithOpacityFade)
```